base-server/index.js

```diff
@@ -27,7 +27,12 @@ function optionError (msg) {
 class BaseServer {
   constructor (opts = {}) {
     this.options = opts
-    this.reporter = this.options.reporter || function () {}
+    this.reporter = function (type, details) {
+      this.emitter.emit('report', type, details)
```

Comment:
I chose to always emit event, cause there are cases when we want to listen for reports, but don't want them printed to logs

Reply:
Why do we still need `this.reporter` and `opts.reporter`?

Reply:
Fair remark

Reply:
We still use reporter inside app, e.g. https://github.com/logux/server/pull/101/files#diff-2a339baf3d65ca89f18b2d3a81f4634fR105 and in server

Reply:
I can't open a link from the phone. What is the file name and line?

Reply:
base-server/index.js 105 line

Reply:
Why we can’t replace it to:

```js
this.emitter.emit('report', type, details)
```

And all `options.reporters` usage can be replaced to `server.on('report'` (if we used it for internal code or dirty hacks).

Reply:
Sounds reasonable